pinot-compatibility-verifier/src/main/java/org/apache/pinot/compat/QueryOp.java

```diff
@@ -72,16 +75,44 @@ public void setExpectedResultsFileName(String expectedResultsFileName) {
     _expectedResultsFileName = expectedResultsFileName;
   }
 
+  public Boolean getShouldCompareNumEntriesScannedInFilter() {
+    return _shouldCompareNumEntriesScannedInFilter;
+  }
+
+  public void setShouldCompareNumEntriesScannedInFilter(Boolean shouldCompareNumEntriesScannedInFilter) {
+    _shouldCompareNumEntriesScannedInFilter = shouldCompareNumEntriesScannedInFilter;
+  }
+
+  public Boolean getShouldCompareNumConsumingSegmentsQueried() {
+    return _shouldCompareNumConsumingSegmentsQueried;
+  }
+
+  public void setShouldCompareNumConsumingSegmentsQueried(boolean shouldCompareNumConsumingSegmentsQueried) {
+    _shouldCompareNumConsumingSegmentsQueried = shouldCompareNumConsumingSegmentsQueried;
+  }
+
+  public Boolean getShouldRunAllGenerations() {
+    return _shouldRunAllGenerations;
+  }
+
+  public void setShouldRunAllGenerations(Boolean shouldRunAllGenerations) {
+    _shouldRunAllGenerations = shouldRunAllGenerations;
+  }
+
   @Override
   boolean runOp(int generationNumber) {
     System.out.println("Verifying queries in " + _queryFileName + " against results in " + _expectedResultsFileName);
     try {
-      for (int i = 1; i <= generationNumber; i++) {
-        if (!verifyQueries(i)) {
-          return false;
+      if (getShouldRunAllGenerations()) {
```

Comment:
I don't understand why we should not be running all generations. 

If we delete and create the table everytime, then it need not be added in compatibility test. The whole idea of compatibility testing is that we can work with tables in which data was added in previous versions, or, while downgrading, the data was added in new versions.

Reply:
Got it. Thanks for pointing out. Then for the index/partition validation relying on the metadata there should be some re-design. I have something in mind and is working on it.